modules/charon-core/src/main/java/org/wso2/charon3/core/schema/AbstractValidator.java

```diff
@@ -113,6 +117,14 @@ private static void validateSCIMObjectForRequiredSubAttributes(AbstractAttribute
                             }
                         }
                     }
+
+                    // Check for canonical attributes in groups.
+                    List<String> canonicalValues = subAttributeSchema.getCanonicalValues();
+                    if (!canonicalValues.isEmpty()) {
```

Comment:
If the `canonicalValues` returned null, won't this throw a null pointer exception? Can we use `CollectionUtils.isNotEmpty`?